wandb/sdk/launch/agent/agent.py

```diff
@@ -421,6 +426,30 @@ async def finish_thread_id(
         """Removes the job from our list for now."""
         with self._jobs_lock:
             job_and_run_status = self._jobs[thread_id]
+
+        try:
+            if hasattr(job_and_run_status, "additional_objects_namespace"):
+                get_module(
+                    "kubernetes_asyncio",
+                    required="Managing additional objects in a job requires the kubernetes_asyncio package. Please install it with `pip install wandb[launch]`.",
+                )
+
+                import kubernetes_asyncio
+                from kubernetes_asyncio.client import CoreV1Api
+
+                namespace = job_and_run_status.additional_objects_namespace
```

Comment:
I don't understand why we're making these in another namespace, is this a requirement somewhere?